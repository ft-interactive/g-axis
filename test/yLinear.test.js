/**
 * @file
 * Test suite for yLinear
 */

import pretty from "pretty";

jest.setTimeout(20000);

beforeAll(global.build("yLinear"));
beforeEach(global.start);

test("left-aligned, default scales", async () => {
    await global.page.evaluate(async () => {
        const sharedConfig = {
            source: false,
            subtitle: "Left-aligned, default scales",
            title: "yLinear test"
        };

        const svg = window.d3.select(document.querySelector("svg"));
        const currentFrame = window.gChartframe.webFrameMDefault(sharedConfig);

        // Set up the chart frame
        svg.call(currentFrame);

        // Instantiate yLinear
        const yAxis = window
            .yLinear()
            .plotDim([
                currentFrame.dimension().width,
                currentFrame.dimension().height
            ])
            .tickSize(currentFrame.dimension().width)
            .domain([0, 200])
            .align("left")
            .rem(currentFrame.rem())
            .range([currentFrame.dimension().height, 0])
            .frameName("webFrameMDefault");

        // Set up yAxis
        currentFrame.plot().call(yAxis);

        // Get newly-calculated margin value
        const newMargin = yAxis.labelWidth() + currentFrame.margin().left;

        // Use newMargin redefine the new margin and range of xAxis
        currentFrame.margin({ left: newMargin });

        // Translate axis from the left
        yAxis
            .yLabel()
            .attr(
                "transform",
                `translate(${Math.floor(
                    yAxis.tickSize() - yAxis.labelWidth()
                )}, 0)`
            );

        // Call parent container to update positioning
        svg.call(currentFrame);
    });

    const bodyHTML = await global.page.evaluate(
        () => document.documentElement.innerHTML
    );
    expect(pretty(bodyHTML)).toMatchSnapshot();
});

test("right-aligned, default scales", async () => {
    await global.page.evaluate(async () => {
        const sharedConfig = {
            source: false,
            subtitle: "Right-aligned, default scales",
            title: "yLinear test"
        };

        const svg = window.d3.select(document.querySelector("svg"));
        const currentFrame = window.gChartframe.webFrameMDefault(sharedConfig);

        // Set up the chart frame
        svg.call(currentFrame);

        // Instantiate yLinear
        const yAxis = window
            .yLinear()
            .plotDim([
                currentFrame.dimension().width,
                currentFrame.dimension().height
            ])
            .tickSize(currentFrame.dimension().width - currentFrame.rem())
            .domain([0, 200])
            .rem(currentFrame.rem())
            .align("right")
            .range([currentFrame.dimension().height, 0])
            .frameName("webFrameMDefault");

        // Set up yAxis
        currentFrame.plot().call(yAxis);

        // Get newly-calculated margin value
        const newMargin = yAxis.labelWidth() + currentFrame.margin().right;

        // Use newMargin redefine the new margin and range of xAxis
        currentFrame.margin({ right: newMargin });

        // Call parent container to update positioning
        svg.call(currentFrame);
    });

    const bodyHTML = await global.page.evaluate(
        () => document.documentElement.innerHTML
    );
    expect(pretty(bodyHTML)).toMatchSnapshot();
});
