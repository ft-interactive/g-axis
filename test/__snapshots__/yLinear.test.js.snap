// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`left-aligned, default scales 1`] = `
"<g class=\\"highlights\\"></g>
<g class=\\"axis yAxis\\" fill=\\"none\\" font-size=\\"10\\" font-family=\\"sans-serif\\" text-anchor=\\"end\\" transform=\\"transform(645, 0)\\">
  <g class=\\"tick baseline\\" opacity=\\"1\\" transform=\\"transform(0, 277)\\">
    <line stroke=\\"#000\\" x2=\\"-645\\" id=\\"webFrameMDefaultyTick\\"></line><text fill=\\"#000\\" x=\\"-648\\" dy=\\"0.32em\\" id=\\"webFrameMDefaultyLabel\\">0</text>
  </g>
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"transform(0, 208)\\">
    <line stroke=\\"#000\\" x2=\\"-645\\" id=\\"webFrameMDefaultyTick\\"></line><text fill=\\"#000\\" x=\\"-648\\" dy=\\"0.32em\\" id=\\"webFrameMDefaultyLabel\\">50</text>
  </g>
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"transform(0, 139)\\">
    <line stroke=\\"#000\\" x2=\\"-645\\" id=\\"webFrameMDefaultyTick\\"></line><text fill=\\"#000\\" x=\\"-648\\" dy=\\"0.32em\\" id=\\"webFrameMDefaultyLabel\\">100</text>
  </g>
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"transform(0, 69)\\">
    <line stroke=\\"#000\\" x2=\\"-645\\" id=\\"webFrameMDefaultyTick\\"></line><text fill=\\"#000\\" x=\\"-648\\" dy=\\"0.32em\\" id=\\"webFrameMDefaultyLabel\\">150</text>
  </g>
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"transform(0, 0)\\">
    <line stroke=\\"#000\\" x2=\\"-645\\" id=\\"webFrameMDefaultyTick\\"></line><text fill=\\"#000\\" x=\\"-648\\" dy=\\"0.32em\\" id=\\"webFrameMDefaultyLabel\\">200</text>
  </g>
</g>"
`;

exports[`right-aligned, default scales 1`] = `
"<g class=\\"highlights\\"></g>
<g class=\\"axis yAxis\\" fill=\\"none\\" font-size=\\"10\\" font-family=\\"sans-serif\\" text-anchor=\\"start\\">
  <g class=\\"tick baseline\\" opacity=\\"1\\" transform=\\"transform(0, 277)\\">
    <line stroke=\\"#000\\" x2=\\"625\\" id=\\"webFrameMDefaultyTick\\"></line><text fill=\\"#000\\" x=\\"628\\" dy=\\"0.32em\\" transform=\\"transform(30, 0)\\" id=\\"webFrameMDefaultyLabel\\">0</text>
  </g>
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"transform(0, 208)\\">
    <line stroke=\\"#000\\" x2=\\"625\\" id=\\"webFrameMDefaultyTick\\"></line><text fill=\\"#000\\" x=\\"628\\" dy=\\"0.32em\\" transform=\\"transform(30, 0)\\" id=\\"webFrameMDefaultyLabel\\">50</text>
  </g>
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"transform(0, 139)\\">
    <line stroke=\\"#000\\" x2=\\"625\\" id=\\"webFrameMDefaultyTick\\"></line><text fill=\\"#000\\" x=\\"628\\" dy=\\"0.32em\\" transform=\\"transform(30, 0)\\" id=\\"webFrameMDefaultyLabel\\">100</text>
  </g>
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"transform(0, 69)\\">
    <line stroke=\\"#000\\" x2=\\"625\\" id=\\"webFrameMDefaultyTick\\"></line><text fill=\\"#000\\" x=\\"628\\" dy=\\"0.32em\\" transform=\\"transform(30, 0)\\" id=\\"webFrameMDefaultyLabel\\">150</text>
  </g>
  <g class=\\"tick\\" opacity=\\"1\\" transform=\\"transform(0, 0)\\">
    <line stroke=\\"#000\\" x2=\\"625\\" id=\\"webFrameMDefaultyTick\\"></line><text fill=\\"#000\\" x=\\"628\\" dy=\\"0.32em\\" transform=\\"transform(30, 0)\\" id=\\"webFrameMDefaultyLabel\\">200</text>
  </g>
</g>"
`;
